src/NBitcoin/ScriptEvaluationContext.cs

```diff
@@ -1797,7 +1797,7 @@ private static int CompareBigEndian(byte[] c1, int ic1, int c1len, byte[] c2, in
         }
 
 
-        public static bool IsValidSignatureEncoding(byte[] sig)
+        public static bool IsValidSignatureEncoding(byte[] sig, bool haveSigHash)
```

Comment:
please default this to true, BTC assumes always a sigh hash, we only remove the sighash if we are checking a POS signed block.